Std/Data/BitVec/Lemmas.lean

```diff
@@ -325,6 +329,14 @@ theorem shiftLeftZeroExtend_eq {x : BitVec w} :
 @[simp] theorem toNat_ushiftRight (x : BitVec n) (i : Nat) :
     (x >>> i).toNat = x.toNat >>> i := rfl
 
+@[simp] theorem toFin_ushiftRight (x : BitVec n) (i : Nat) :
+    (x >>> i).toFin = ⟨x.toNat >>> i, by
+      let ⟨x, lt⟩ := x
+      simp only [BitVec.toNat, Nat.shiftRight_eq_div_pow, Nat.div_lt_iff_lt_mul (Nat.two_pow_pos i)]
+      rw [←Nat.mul_one x]
+      exact Nat.mul_lt_mul_of_lt_of_le' lt (Nat.two_pow_pos i) (Nat.le_refl 1)⟩ := by
+  simp only [HShiftRight.hShiftRight, ushiftRight, BitVec.toFin]
```

Comment:
I'm unsure about this. It feels like we should define shifting on `Fin`, instead of inlining this proof.
I'd say put it in the backlog